src/internal/modules/charge-information/controller.js

```diff
@@ -105,47 +107,106 @@ const postUseAbstractionData = createPostHandler(
   request => routing.getCheckData(request.pre.licence)
 );
 
-const getCheckData = async (request, h) => {
-  const { licence, draftChargeInformation } = request.pre;
+const getInvoiceAccountAddressFromDraft = draftChargeInformation => {
```

Comment:
I would expect we would have something in the session along the lines of:
```
{
  "chargeVersion" : {
    "invoiceAccount" : {
        "id" : "43443-434-43-334-3443-4"
     },
     ...
  }
```
This single invoice account ID should be enough to get the invoice account from the crm via the water service including its currently selected address.

I'm assuming the data in `billingAccount.billingAccount...` has been collected during the "select invoice account" sub-flow, but if a charge version is in review and loaded back from the workflow table by a different user, none of this would be populated.